src/contracts/core/DelegationManager.sol

```diff
@@ -193,162 +228,167 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
+        emit StakerDelegated(staker, operator);
     }
 
     // VIEW FUNCTIONS
+    /**
+     * @notice Getter function for the current EIP-712 domain separator for this contract.
+     * @dev The domain separator will change in the event of a fork that changes the ChainID.
+     */
+    function domainSeparator() public view returns (bytes32) {
+        if (block.chainid == ORIGINAL_CHAIN_ID) {
+            return _DOMAIN_SEPARATOR;
+        }
+        else {
+            return _calculateDomainSeparator();
+        }
+    }
 
     /// @notice Returns 'true' if `staker` *is* actively delegated, and 'false' otherwise.
     function isDelegated(address staker) public view returns (bool) {
         return (delegatedTo[staker] != address(0));
     }
 
-    /// @notice Returns 'true' if `staker` is *not* actively delegated, and 'false' otherwise.
-    function isNotDelegated(address staker) public view returns (bool) {
-        return (delegatedTo[staker] == address(0));
+    /// @notice Returns if an operator can be delegated to, i.e. the `operator` has previously called `registerAsOperator`.
+    function isOperator(address operator) public view returns (bool) {
+        return (_operatorDetails[operator].earningsReceiver != address(0));
     }
 
-    /// @notice Returns if an operator can be delegated to, i.e. it has called `registerAsOperator`.
-    function isOperator(address operator) public view returns (bool) {
-        return (address(delegationTerms[operator]) != address(0));
+    /**
+     * @notice returns the OperatorDetails of the `operator`.
+     * @notice Mapping: operator => OperatorDetails struct
+     */
+    function operatorDetails(address operator) external view returns (OperatorDetails memory) {
+        return _operatorDetails[operator];
+    }
+
+    // @notice Getter function for `_operatorDetails[operator].earningsReceiver`
+    function earningsReceiver(address operator) external view returns (address) {
+        return _operatorDetails[operator].earningsReceiver;
+    }
+
+    // @notice Getter function for `_operatorDetails[operator].delegationApprover`
+    function delegationApprover(address operator) external view returns (address) {
+        return _operatorDetails[operator].delegationApprover;
+    }
+
+    // @notice Getter function for `_operatorDetails[operator].stakerOptOutWindowBlocks`
+    function stakerOptOutWindowBlocks(address operator) external view returns (uint256) {
+        return _operatorDetails[operator].stakerOptOutWindowBlocks;
+    }
+
+    /**
+     * @notice External getter function that mirrors the staker signature hash calculation in the `delegateToBySignature` function
+     * @param staker The signing staker
+     * @param operator The operator who is being delegated
+     * @param expiry The desired expiry time of the staker's signature
+     */
+    function calculateStakerDigestHash(address staker, address operator, uint256 expiry) external view returns (bytes32) {
```

Comment:
why not make public and reuse?

Reply:
we discussed what I think is a good path forward on this offline. I just pushed a commit which you should review:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/97/commits/787596a7f88dd4dbd057b588fefb132162467f2b